deepmd/descriptor/se_a_mask.py

```diff
@@ -0,0 +1,429 @@
+
+        Parameters
+        ----------
+        coord_
+                The coordinate of atoms
+        atype_
+                The type of atoms
+        natoms
+                The number of atoms. This tensor has the length of Ntypes + 2
+                natoms[0]: number of local atoms
+                natoms[1]: total number of atoms held by this processor
+                natoms[i]: 2 <= i < Ntypes+2, number of type i atoms
+        mesh
+                For historical reasons, only the length of the Tensor matters.
+                if size of mesh == 6, pbc is assumed. 
+                if size of mesh == 0, no-pbc is assumed. 
+        input_dict
+                Dictionary for additional inputs
+        reuse
+                The weights in the networks should be reused when get the variable.
+        suffix
+                Name suffix to identify this descriptor
+
+        Returns
+        -------
+        descriptor
+                The output descriptor
+        """
+        #davg = self.davg
+        #dstd = self.dstd
+        davg = None
+        dstd = None
+        with tf.variable_scope('descrpt_attr' + suffix, reuse = reuse) :
+            if davg is None:
+                davg = np.zeros([self.ntypes, self.ndescrpt]) 
+            if dstd is None:
+                dstd = np.ones ([self.ntypes, self.ndescrpt])
+            #t_rcut = tf.constant(np.max([self.rcut_r, self.rcut_a]), 
+            #                     name = 'rcut', 
+            #                     dtype = GLOBAL_TF_FLOAT_PRECISION)
+            t_ntypes = tf.constant(self.ntypes, 
+                                   name = 'ntypes', 
+                                   dtype = tf.int32)
+            t_ndescrpt = tf.constant(self.ndescrpt, 
+                                     name = 'ndescrpt', 
+                                     dtype = tf.int32)            
+            t_sel = tf.constant(self.sel_a, 
+                                name = 'sel', 
+                                dtype = tf.int32)            
+            '''
+            self.t_avg = tf.get_variable('t_avg', 
+                                         davg.shape, 
+                                         dtype = GLOBAL_TF_FLOAT_PRECISION,
+                                         trainable = False,
+                                         initializer = tf.constant_initializer(davg))
+            self.t_std = tf.get_variable('t_std', 
+                                         dstd.shape, 
+                                         dtype = GLOBAL_TF_FLOAT_PRECISION,
+                                         trainable = False,
+                                         initializer = tf.constant_initializer(dstd))
+            '''
+            
+        coord = tf.reshape (coord_, [-1, natoms[1] * 3])
+        box   = tf.reshape (box_, [-1, 9])
+        atype = tf.reshape (atype_, [-1, natoms[1]])
+        mask_matrix = input_dict["mask_matrix"]
```

Comment:
I am not quite understanding here - where does `input_dict["mask_matrix"]` come from? I don't see any related codes

Reply:
That's load from `mask_matrix.npy` in set directory as I used `add_data_requirement('mask_matrix', 1, atomic=True, must=True, high_prec=False)`. As I used this in `EnerForcesMaskLoss` class, the data system will load `mask_matrix.npy` into `input_dict` as the value of key `mask_matrix` automatically

Reply:
`add_data_requirement` is only in `EnerForcesMaskLoss`. Does it mean it must be used along with `EnerForcesMaskLoss`? Or may you also add `add_data_requirement` to `DescrptSeAMask`? (`data_requirement` is a dict so it will not be added twice)

Reply:
OK. I will do this later.